frontend/src/pages/admin/users/ManageUsers.tsx

```diff
@@ -12,55 +11,101 @@ import './ManageUsers.scss';
   const pagedUsers = useSelector<RootState, IPagedItems<IUser>>(state => {
-    return (state.users as IStoreUsersAction).pagedUsers;
+    return (state.users as IUsersState).pagedUsers;
   });
+
+  const pageSize = useSelector<RootState, number>(state => {
+    return (state.users as IUsersState).rowsPerPage;
+  });
+
+  const sort = useSelector<RootState, IUsersSort>(state => {
+    return (state.users as IUsersState).sort;
+  });
+
   const users = useSelector<RootState, IGenericNetworkAction>(
     state => (state.network as any)[actionTypes.GET_USERS] as IGenericNetworkAction,
   );
   useEffect(() => {
     dispatch(getUsersAction(toApiPaginateParams(0, MAX_USERS_PER_PAGE)));
-
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   const tableChange: TableStateChangeFn = (type: string, state: MUIDataTableState) => {
     switch (type) {
-      case 'pagination':
-        break;
-      case 'filter':
-        break;
       case 'sort':
+        const sortColumn = state.columns.find(x => x.sortDirection !== 'none');
+        if (sortColumn) {
+          dispatch(
+            setUsersSort({
+              sortBy: sortColumn.name as any,
+              direction: sortColumn.sortDirection as any,
+            }),
+          );
+        }
+        break;
+      case 'resetFilters':
+      case 'changePage':
+      case 'filterChange':
+      case 'changeRowsPerPage':
+        dispatch(setUsersFilter(state.filterList));
+        dispatch(setUsersPageSize(state.rowsPerPage));
+
+        dispatch(
+          getUsersPaginationAction({
+            page: state.page + 1,
+            quantity: state.rowsPerPage,
+            username: state.filterList[1][0] || '',
+            firstName: state.filterList[2][0] || '',
+            lastName: state.filterList[3][0] || '',
+            email: state.filterList[4][0] || '',
+            isDisabled: accountActiveToBool(state.filterList[5][0]),
+            agency: state.filterList[6][0] || '',
+            position: state.filterList[7][0] || '',
+            role: state.filterList[8][0] || '',
+          }),
+        );
         break;
     }
   };
-  const userList = pagedUsers.items.map(
+
+  let userList = pagedUsers.items.map(
     (u: IUser): IUserRecord => ({
       id: u.id,
       email: u.email,
       username: u.username,
       firstName: u.firstName,
       lastName: u.lastName,
       active: u.isDisabled ? AccountActive.NO : AccountActive.YES,
-      role: u.roles ? u.roles[0].name : 'No role attached',
-      agency: u.agencies ? u.agencies[0].name : 'No agency associated',
+      role: u.roles && u.roles.length > 0 ? u.roles[0].name : 'No role attached',
+      agency: u.agencies && u.agencies.length > 0 ? u.agencies[0].name : 'No agency associated',
       position: u.position,
     }),
   );
 
+  if (sort) {
+    userList = sortBy(userList, u => u[sort.sortBy]);
+    if (sort.direction === 'desc') {
+      userList = userList.reverse();
```

Comment:
This does reverse the order of the list assuming that the sort order of all elements is equal correct?

ie. if I order by position [{position: 'test', name:'doug'}, {position: 'test', name 'frank'}] -> [{position: 'test', name 'frank'}, {position: 'test', name:'doug'}]

Reply:
@devinleighsmith I am not sure if I understand. Even when the values for position are the same, the reverse of the sort will give the correct results